docs/28.API-Tests.md

```
@@ -0,0 +1,605 @@
+
+subgraph subGraph0 ["External Dependencies"]
+    ArWikiCats
+    LogsDB
+    LogsBot
+end
+```
+
+**Common Mock Patterns:**
+
+1. **Library Function Mocking:** [tests/test_api.py L203-L214](https://github.com/ArWikiCats/ArWikiCatsWeb/blob/88f42d13/tests/test_api.py#L203-L214) ``` with patch("src.app.routes.api.resolve_arabic_category_label") as mock_resolve:     mock_resolve.return_value = "تصنيف:اختبار" ```
+2. **Database Function Mocking:** [tests/test_logs_bot.py L39-L48](https://github.com/ArWikiCats/ArWikiCatsWeb/blob/88f42d13/tests/test_logs_bot.py#L39-L48) ```python @patch("src.app.logs_bot.logs_db") def test_view_logs_returns_dict_with_required_keys(self, mock_logs_db):     mock_logs_db.get_logs.return_value = [] ```
```

markdownlint-cli2 (0.20.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)